### PR TITLE
Solution for issue #6 - Gravatar. (team 4)

### PR DESCRIPTION
**Pull request by byteme**
Added Gravtar support for piplmesh.
- Image template in gravatar.html (frontend/templates)
- Template tag gravatar.py (frontend/templatetags)
- Added settings for default gravatar protocol (https/http)

Update:
- Added default avatar support

### CODE REVIEW

**Comment by byteme** [comment]
Updated pull request (forgot to add default avatar support last time).

File: piplmesh/frontend/templatetags/__init__.py [hunk 1]
```diff
@@ -0,0 +1 @@
+__author__ = 'David'
```
Comment (mitar) [inline piplmesh/frontend/templatetags/__init__.py:1]: Don't do this.

File: piplmesh/frontend/templatetags/gravatar.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
+from django import template
+from django.conf import settings
+from django.contrib.sites import models as sites_models
+import urllib, hashlib
```
Comment (mitar) [inline piplmesh/frontend/templatetags/gravatar.py:4]: Import order. Built-in modules are first (in a separate group).

File: piplmesh/frontend/templatetags/gravatar.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    Sample usage:
+        {% load gravatar %}
+        {% gravatar user_obj %}
+    """
+
+    protocol = 'https' if settings.GRAVATAR_HTTPS_DEFAULT else 'http'
+
+    # Build default gravatar URL
+    default_avatar_url = 'http://%(domain)s%(static)spiplmesh/images/%(default_avatar)s' % {
+        'domain': sites_models.Site.objects.get_current().domain,
```
Comment (mitar) [inline piplmesh/frontend/templatetags/gravatar.py:22]: What if sites framework is not installed/used?

File: piplmesh/frontend/templatetags/gravatar.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        'static': settings.STATIC_URL,
+        'default_avatar': default_avatar
```
Comment (mitar) [inline piplmesh/frontend/templatetags/gravatar.py:24]: Comma at the end.

File: piplmesh/frontend/templatetags/gravatar.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    # Build gravatar URL
+    gravatar_url = '%(protocol)s://gravatar.com/avatar/%(email_hash)s?s=%(size)s&d=%(default_avatar_url)s' % {
+        'protocol': protocol,
+        'email_hash': hashlib.md5(user.email.lower()).hexdigest(),
+        'size': size,
+        'default_avatar_url': default_avatar_url
```
Comment (mitar) [inline piplmesh/frontend/templatetags/gravatar.py:32]: Comma at the end.

File: piplmesh/frontend/templatetags/gravatar.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    return {
+        'gravatar': {
+            'url': gravatar_url,
+            'size': size,
+            'alt': user.username    # later should be changed into first name + last name for SEO purposes
```
Comment (mitar) [inline piplmesh/frontend/templatetags/gravatar.py:39]: Good idea this about SEO. But will our portal be accessible from search engines? How Facebook managed to get user profiles into the Google, but when you visit the page not logged in you have just one empty login page?

File: piplmesh/frontend/views.py [hunk 7]
```diff
@@ -1,5 +1,7 @@
 from django.views import generic as generic_views
 
+from django.contrib.auth.models import User
```
Comment (mitar) [inline piplmesh/frontend/views.py:3]: Import only modules.

**Comment by mitar** [comment]
Closed in favor of #46.

File: piplmesh/frontend/templatetags/__init__.py (hunk 1 above)
Comment (byteme) [inline piplmesh/frontend/templatetags/__init__.py:1]: Forgot to delete it. It's auto create by pycharm.

File: piplmesh/frontend/templatetags/gravatar.py (hunk 6 above)
Comment (byteme) [inline piplmesh/frontend/templatetags/gravatar.py:39]: Well parts of the portal will be definitely w/o authorization, how much of the user profile will depend on user settings.

Comment (mitar) [inline piplmesh/frontend/templatetags/gravatar.py:39]: OK. You are right. But we have also one additional speciality: our portal will be accessible inside the wireless network. Will we want to allow it also to be accessible outside? ;-)

**Pull request closed without merging**